# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1293968045 -3600
#      Sun Jan 02 12:34:05 2011 +0100
# Node ID 9f4321024fbf73f294e3e2f27dd52bfd119c4239
# Parent  da51bc237526b7659c1d4b006f3e296e449ff23f
make insumeric return false on sparse logicals

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2011-01-02  Jaroslav Hajek  <highegg@gmail.com>
+
+	* ov-bool-sparse.h (octave_sparse_bool_matrix::is_numeric_type): New
+	virtual method override.
+
 2010-12-31  Rik  <octave@nomad.inbox5.com>
 
 	* toplev.cc (system): Add additional calling form "shell_cmd" to
 	documentation.
 
 2010-12-31  Rik  <octave@nomad.inbox5.com>
 
 	* data.cc (islogical, isinteger, iscomplex, isfloat, isempty,
diff --git a/src/ov-bool-sparse.h b/src/ov-bool-sparse.h
--- a/src/ov-bool-sparse.h
+++ b/src/ov-bool-sparse.h
@@ -90,16 +90,18 @@ public:
   builtin_type_t builtin_type (void) const { return btyp_bool; }
 
   bool is_bool_matrix (void) const { return true; }
 
   bool is_bool_type (void) const { return true; }
 
   bool is_real_type (void) const { return true; }
 
+  bool is_numeric_type (void) const { return false; }
+
   double double_value (bool = false) const;
 
   double scalar_value (bool frc_str_conv = false) const
     { return double_value (frc_str_conv); }
 
   Matrix matrix_value (bool = false) const;
 
   NDArray array_value (bool = false) const;
